CLAUDE.md

```diff
@@ -0,0 +1,149 @@
+2. Use error wrapping with `fmt.Errorf("context: %w", err)` for additional context
+   - Ensure that `%w` is used for error wrapping, *not* `%v`
+
+### 3.2. Code Documentation
+
+1. Write docs for all exported functions/types in production code
+2. Write docs for unexported functions/types if they contain non-trivial logic. A good rule of thumb: if you can't understand everything
+   there is to know about a function/type by its *name*, you should write a doc.
+3. Function/type docs should NOT simply be a rephrasing of the function/type name.
+   - E.g. the doc for `computeData` should NOT be "Computes the data".
+4. Function docs should consider the following helpful information, if relevant:
+   - What are the inputs?
+   - Are there any restrictions on what the input values are permitted to be?
+   - What is returned in the standard case?
+   - What is returned in the error case(s)?
+   - What side effects does calling the function have?
+   - Are there any performance implications that users should be aware of?
+
+### 3.3. Doc Files
+
+1. **Hierarchical organization**: Hierarchical numbering for sections makes referencing easier.
+2. **Tabular format for key facts**: Tables are helpful for understanding data at a glance: use them where appropriate.
+3. **Use Links**: Links are very helpful to assist a human navigating through the codebase.
+   - IMPORTANT: double check that links aren't broken after making changes to doc files. Similarly, if documentation
+   contains links directly to code, make sure that code changes are paired with the corresponding doc updates.
+
+---
+
+## 4. Directory-Specific CLAUDE.md Files
+
+1. **Always check for `CLAUDE.md` files in specific directories** before working on code within them. These files contain targeted context.
+2. If a directory's `CLAUDE.md` is outdated or incorrect, **update it**.
+3. If you make significant changes to a directory's structure, patterns, or critical implementation details, **document these in its `CLAUDE.md`**.
+4. If a directory lacks a `CLAUDE.md` but contains complex logic or patterns worth documenting for AI/humans, **suggest creating one**.
+5. Use `@` annotation within CLAUDE.md files to automatically load in helpful context, e.g. `@docs/submoduleDocs`. These imports will be automatically
+   processed whenever the `CLAUDE.md` file is read.
+6. If there is domain-specific terminology relevant to a directory, consider adding a small glossary of terms.
+
+---
+
+## 5. Common pitfalls
+
+1. Forgetting to run `go mod tidy` after adding new dependencies.
+2. Not linting before committing code.
+3. Wrong working directory when running commands (ensure you're in project root).
+4. Large AI refactors in a single commit.
+5. Delegating test/spec writing entirely to AI (can lead to false confidence).
+
+---
+
+## 6. Files to NOT modify
+
+These files and directories should generally not be modified without explicit permission:
+
```

Comment:
Should we explicitly state bindings?

Reply:
03467b6dd00bda708434547ce249a0a4fdabc93a